calicoctl.py

```diff
@@ -302,8 +302,10 @@ def node(ip, force_unix_socket, node_image, ip6=""):
 
         # Set the docker daemon to listen on the docker.real.sock by updating
         # the config, clearing old sockets and restarting.
-        socket_config_exists = \
+        socket_config_exists = (
+            os.path.isfile(DOCKER_DEFAULT_FILENAME) and
             DOCKER_OPTIONS in open(DOCKER_DEFAULT_FILENAME).read()
+        )
```

Comment:
I don't think this works.  If DOCKER_DEFAULT_FILENAME doesn't exist, then we can't support the --force-unix-socket option.  Writing out that file and restarting docker will not have any effect on Docker if it isn't set to look for its config in that file!

We should just error out if we can't find the file and they've used --force-unix-socket